marklogic-data-hub-central/ui/src/components/entities/matching/ruleset-single-modal/ruleset-single-modal.tsx

```diff
@@ -76,40 +100,78 @@ const MatchRulesetModal: React.FC<Props> = (props) => {
       case 'exact':
       case 'reduce':
       case 'zip':
-        let propertyName = selectedProperty || '';
-
-        let matchRule: MatchRule = {
-          entityPropertyPath: propertyName,
-          matchType: matchType,
-          options: {}
-        }
+        {
+          let propertyName = selectedProperty || '';
 
-        let matchRuleset: MatchRuleset = {
-          name: propertyName,
-          weight: 0,
-          matchRules: [matchRule]
+          let matchRule: MatchRule = {
+            entityPropertyPath: propertyName,
+            matchType: matchType,
+            options: {}
+          }
+  
+          let matchRuleset: MatchRuleset = {
+            name: propertyName,
+            weight: 0,
+            matchRules: [matchRule]
+          }
+  
+          //check for duplicate
+          let propertyRulesets = curationOptions.activeStep.stepDefinition['matchRulesets'].filter(matchRuleset => matchRuleset['name'] === propertyName)
+          if (propertyRulesets.length > 0) {
```

Comment:
@ryanjdew @brucean52 @wooldridge I think the backend should validate this. But I also don't see any requirement in 5254 about prohibiting duplicates, nor what defines a duplicate.

Reply:
I confirmed with @ryanjdew that we shouldn't allow duplicate rulesets.

Reply:
But what is the logic for defining "duplicate"? That's not in the JIRA ticket either. This logic only is applied on rulesets whose matchRules array has a length of 1. What if it's 2? 

My main point is - since the JIRA ticket says nothing about duplicates, there shouldn't be any code in this PR for duplicates. Otherwise, how does whomever does QA on this know if the duplicate logic is correct?

Reply:
If it's 2 then it isn't a duplicate. It's a ruleset for multiple properties and not a single one. 

Error handling isn't described in all tickets. Should it be a separate ticket?

Reply:
We only want to write code in support of requirements written in JIRA tickets. I assume that this duplicate logic would be relevant regardless of how the step is being saved - i.e. it should be applied when saving a step via Gradle. So the logic should be in the backend. Since 5254 seems to be a UI-only ticket, then a separate ticket makes sense.